File: app/src/main/java/de/bornholdtlee/snh/model/converter/StringListConverter.java
package de.bornholdtlee.snh.model.converter;

import android.text.TextUtils;

import java.util.ArrayList;
import java.util.List;

import io.objectbox.converter.PropertyConverter;

public class StringListConverter implements PropertyConverter<List<String>, String> {

    private static final String SEPERATOR = ";";

    @Override
    public List<String> convertToEntityProperty(String databaseValue) {
        String[] split = databaseValue.split(SEPERATOR);
        List<String> result = new ArrayList<>();
        for (String aSplit : split) {
            result.add(aSplit);
        }
        return result;
    }

    @Override
    public String convertToDatabaseValue(List<String> strings) {
        String result = "";
        for (String integer : strings) {
            if (!TextUtils.isEmpty(result)) {
                result += SEPERATOR;
            }
            result += integer;
        }
        return result;
    }

}

File: app/src/main/java/de/bitb/astroskop/model/converter/StringListConverter.kt
package de.bitb.astroskop.model.converter

import android.text.TextUtils

import java.util.ArrayList

import io.objectbox.converter.PropertyConverter

class StringListConverter : PropertyConverter<List<String>, String> {

    override fun convertToEntityProperty(databaseValue: String): List<String> {
        val split = databaseValue.split(SEPERATOR.toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        val result = ArrayList<String>()
        for (aSplit in split) {
            result.add(aSplit)
        }
        return result
    }

    override fun convertToDatabaseValue(strings: List<String>): String {
        var result = ""
        for (integer in strings) {
            if (!TextUtils.isEmpty(result)) {
                result += SEPERATOR
            }
            result += integer
        }
        return result
    }

    companion object {

        private val SEPERATOR = ";"
    }

}
